articles/logging-with-winston/index.md

```diff
@@ -0,0 +1,575 @@
+
+Here is an example of a Morgan logger:
+
+```js
+const express = require('express')
+const fs = require('fs')
+const morgan = require('morgan')
+const path = require('path')
+const app = express()
+
+// Create a write stream (in append mode)
+const accessLogStream = fs.createWriteStream(path.join(__dirname, 'access.log'), { flags: 'a' })
+
+// Setup the logger
+app.use(morgan('combined', { stream: accessLogStream }))
+
+app.get('/', (req, res) => {
+    res.send('Hello World!')
+})
+app.listen(3000, () => {
+    console.log("Server Listening on port 3000");
+})
+```
+
+[*Code Source*](https://www.npmjs.com/package/morgan#write-logs-to-a-file)
+
+Run the app and hit http://localhost:3000/ on your browser. Morgan will record the following log on the access.log file.
+
+```js
+::1 - - [12/Nov/2020:09:19:46 +0000] "GET / HTTP/1.1" 304 - "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.183 Safari/537.36"
+```
+
+2. Bunyan
+
+[Bunyan](https://www.npmjs.com/package/bunyan) records log entries in JSON format. The format is automatically generated with essential fields such as timestamp and hostname.
+
+Like the Winston logger, create a logger instance and log your message.
+
+```js
+const Bunyan = require("Bunyan");
+const log = bunyan.createLogger({ name: "myapp" });
+log.info("hi");
+log.warn({ lang: "fr" }, "au revoir");
+```
+
+Output:
+
+```js
+{"name":"myapp","hostname":"Doe","pid":14244,"level":30,"msg":"hi","time":"2020-11-12T08:22:41.398Z","v":0}
+{"name":"myapp","hostname":"Doe","pid":14244,"level":40,"lang":"fr","msg":"au revoir","time":"2020-11-12T08:22:41.4
+00Z","v":0}
+```
+
+3. Log4js
+
+The following [Log4js](https://github.com/log4js-node/log4js-node) example will record error logs to both the file (log4.log) and the console appenders.
+
+```js
+// Require Log4js
+const log4js = require('log4js');
+
+// Logger configuration
+log4js.configure({
+    appenders: {
+        fileAppender: { type: 'file', filename: './logs/log4.log' },
+        console: { type: 'console' }
+    },
+    categories: {
+        default: { appenders: ['fileAppender', 'console'], level: 'error' }
+    }
+});
+
+// Create the logger
+const logger = log4js.getLogger();
+
+// Log some messages
+logger.trace('Trace, log4js!');
+logger.debug('Debug, log4js!');
+logger.info('Hello, log4js!');
+logger.warn('Heads up, log4js!');
+logger.error('Danger, log4js!');
+logger.fatal('Fatal, log4js!');
+```
+
+Output:
+
+```js
+[2020-11-12T11:27:21.278] [ERROR] default - Danger, log4js!
+[2020-11-12T11:27:21.291] [FATAL] default - Fatal, log4js!
+```
+
+### Final Notes
+Logging is the best approach to adopt for your production application. But before logging consider:
+
+#### When to do Logging
+When an application is at the production level, you can adapt many logging approaches depending on your app usage. Logging can combine one or more logging instances. Some of the standard logging instances include:
```

Comment:
When an application is up and running, there are many different logging approaches you could use, depending on your app's functionality. **- Flow edit**